lib/resque/scheduler.rb

```diff
@@ -419,11 +410,43 @@ def logger
 
       private
 
+      def queue_info_from(config)
+        klass_name = config['class'] || config[:class]
+        begin
+          klass = Resque::Scheduler::Util.constantize(klass_name)
+        rescue NameError
+          klass = klass_name
+        end
+
+        queue_name = config['queue'] || config[:queue] || Resque.queue_from_class(klass)
+        [klass, klass_name, queue_name]
+      end
+
+      def in_progress?(queue_name)
+        currently_processing = Resque::Worker.working.map(&:job).any? do |job|
+          job['queue'] == queue_name.to_s
+        end
+        return true if currently_processing
+
+        Resque.peek(queue_name, 0, 5000).any?
```

Comment:
Trying to figure out, if there's either a worker still working on the scheduled job or there's already one in the queue.
Granted, it's awkward. I updated the PR.